File: 575. Distribute Candies.java
public class Solution {
    public int distributeCandies(int[] candies) {
        int output = 0;
        int s = (candies.length)/2;
        Arrays.sort(candies);
        HashSet<Integer> hs = new HashSet<Integer>();
        for (int i = 0; i < candies.length; i++) {
            hs.add(candies[i]);
        }
        if (hs.size() > s) {
            return s;
        }
        output = hs.size();
        return output;
    }
}

File: 575. Distribute Candies.py
from collections import Counter
class Solution:
    def distributeCandies(self, candies):
        """
        :type candies: List[int]
        :rtype: int
        """
        output = len(candies) // 2 
        candy = set(candies)
        candy_count = len(candy)
        
        return output if candy_count > output else candy_count
